torch/_jit_internal.py

```diff
@@ -140,7 +140,13 @@ def can_compile_class(cls):
     # be compiled and is probably a builtin / bound from C
     if is_ignored_fn(cls):
         return False
-    fns = [getattr(cls, name) for name in cls.__dict__ if inspect.isroutine(getattr(cls, name))]
+    names = cls.__dict__
+    # '__abstractmethods__' errors when accessed as an AttributeError, so
+    # exit before calling getattr below. This can occur in a type object or class
+    # with an abstract method, in either case we cannot compile it as a class
+    if '__abstractmethods__' in names:
+        return False
+    fns = [getattr(cls, name) for name in names if inspect.isroutine(getattr(cls, name))]
```

Comment:
You can avoid special casing `type` here by changing the `getattr` to `getattr(cls, name, None)`

Reply:
Or better yet change this whole thing to this which doesn't have the problem of accessing the attribute at all:

```python
members = inspect.getmembers(cls)
members = map(lambda x: x[1], members)
fns = filter(inspect.isroutine, members)
```